dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/GlobalTaskDispatchWaitingQueueLooper.java

```diff
@@ -65,24 +66,20 @@ public void run() {
     }
 
     void doDispatch() {
-        final ITaskExecutionRunnable taskExecutionRunnable = globalTaskDispatchWaitingQueue.takeTaskExecuteRunnable();
-        final TaskInstance taskInstance = taskExecutionRunnable.getTaskInstance();
-        try {
-            final TaskExecutionStatus status = taskInstance.getState();
-            if (status != TaskExecutionStatus.SUBMITTED_SUCCESS && status != TaskExecutionStatus.DELAY_EXECUTION) {
```

Comment:
Why you remove this? Are you clear on what this code does? After your removed, the task instance cannot be removed after killed.

Reply:
Thank you for your concern. I have moved the status check to the WorkerGroupTaskDispatcher, which should prevent the task instance from not being removable after being killed. However, I understand that moving it to the global queue for an early check is also a viable option. I appreciate your feedback and will keep this in mind for further improvements.

Reply:
Thank you for bringing this up. I have moved the status check to the WorkerGroupTaskDispatcher. I believe this change should ensure that the task instance can be removed after being killed.   Placing the status check in the global queue could indeed be a better approach. I’m happy to explore that option further if you think it would improve the process.

Reply:
> Thank you for bringing this up. I have moved the status check to the WorkerGroupTaskDispatcher. I believe this change should ensure that the task instance can be removed after being killed. Placing the status check in the global queue could indeed be a better approach. I’m happy to explore that option further if you think it would improve the process.

 After some consideration, I have decided to add status checks in the global queue as well.

Reply:
Remove the close logic for WorkerGroupTaskDispatcher. Since this scenario is rare and the logic is simplified for now, it will be left as a TODO.